### PR TITLE
Test for editCommand

### PR DESCRIPTION
**Pull request by HongRay**
Test for editcommand

First few test written for EditCommand, not 100% coverage.
Will think of what other test to write in the future.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -165,6 +165,15 @@ public EditPersonDescriptor(EditPersonDescriptor toCopy) {
...
         }
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public EditPersonDescriptor(Person toCopy) {
+            assert toCopy != null;
+            this.entryList = toCopy.getList();
+            this.tags = toCopy.getTags();
+        }
```
Comment (Xuyan0518) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:176]: Nice job! You are so clever to integrate defensive coding into your code.

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -97,4 +128,23 @@ public static void assertCommandFailure(Command command, Model actualModel, Stri
...
+     * {@code model}'s address book.
+     */
+    public static void showPersonAtIndex(Model model, Index targetIndex) {
+        assertTrue(targetIndex.getZeroBased() < model.getFilteredPersonList().size());
+
+        Person person = model.getFilteredPersonList().get(targetIndex.getZeroBased());
+        Map<String, String> catDescription = new HashMap<>();
+        catDescription.put("Name", "Alice");
+        Set<Tag> tags = person.getTags();
+        Set<String> tagSet = tags.stream()
```
Comment (Xuyan0518) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:142]: The method is well-written. LGTM.

**Review by Cedricaca — COMMENTED** [review]
nice tests

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -165,6 +165,15 @@ public EditPersonDescriptor(EditPersonDescriptor toCopy) {
             this.category = toCopy.getCategory();
             this.description = toCopy.getDescription();
         }
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
```
Comment (Cedricaca) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:170]: not bad sia

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 4]
```diff
@@ -58,5 +58,8 @@ public int hashCode() {
     public String toString() {
         return '[' + tagName + ']';
     }
+    public String getTagName() {
+        return tagName;
```
Comment (Cedricaca) [inline src/main/java/seedu/address/model/tag/Tag.java:62]: oioi so nice function

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 5]
```diff
@@ -97,4 +128,23 @@ public static void assertCommandFailure(Command command, Model actualModel, Stri
         assertEquals(expectedFilteredList, actualModel.getFilteredPersonList());
     }
 
+    /**
+     * Updates {@code model}'s filtered list to show only the person at the given {@code targetIndex} in the
+     * {@code model}'s address book.
+     */
+    public static void showPersonAtIndex(Model model, Index targetIndex) {
```
Comment (Cedricaca) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:135]: i really like this function

**Review by Cedricaca — APPROVED** [review]
i love your style

**Pull request merged**